core/variables.js

```diff
@@ -227,47 +227,57 @@ Blockly.Variables.generateUniqueName = function(workspace) {
  * Create a new variable on the given workspace.
  * @param {!Blockly.Workspace} workspace The workspace on which to create the
  *     variable.
- * @return {null|undefined|string} An acceptable new variable name, or null if
- *     change is to be aborted (cancel button), or undefined if an existing
- *     variable was chosen.
+ * @param {function(null|undefined|string)} opt_callback The callback that returns
```

Comment:
Also need to annotate this as optional.  I think that will look like {function(null|undefined|string)=}